src/random.cpp

```diff
@@ -192,20 +192,38 @@ uint64_t GetRdSeed() noexcept
 #elif defined(__aarch64__) && defined(HWCAP2_RNG)
 
 bool g_rndr_supported = false;
+bool g_rndrrs_supported = false;
+
+bool VerifyRNDRRS()
+{
+    uint8_t ok;
+    uint64_t test;
+    int max_retries = 1000;
+    do {
+        __asm__ volatile("mrs %0, s3_3_c2_c4_1; cset %w1, ne;"
+                         : "=r"(test), "=r"(ok)::"cc");
+        if (ok) return true;
+        __asm__ volatile("yield");
```

Comment:
yielding a 1000 timeshares to check if the RNG is working is probably overkill, i'd say if it fails like, 10 times in a row that's already enough indication we shouldn't use it?